src/components/Dialog/CollectionsDialog.js

```diff
@@ -0,0 +1,225 @@
+    description: 'Enter name for this new collection.',
+    submitButton: 'Submit',
+    showCancelButton: true,
+    showTextInput: true,
+    useDupLabel: false,
+  },
+  [COLLECTIONS_DIALOG_ACTIONS.RENAME]: {
+    action: COLLECTIONS_DIALOG_ACTIONS.RENAME,
+    title: 'Rename Collection',
+    description: 'Enter name for this new collection.',
+    submitButton: 'Rename',
+    showCancelButton: true,
+    showTextInput: true,
+    useDupLabel: true,
+  },
+  [COLLECTIONS_DIALOG_ACTIONS.DUPLICATE]: {
+    action: COLLECTIONS_DIALOG_ACTIONS.DUPLICATE,
+    title: 'Duplicate Collection',
+    description: 'Enter name for this new collection.',
+    submitButton: 'Duplicate',
+    showCancelButton: true,
+    showTextInput: true,
+    useDupLabel: true,
+  },
+  [COLLECTIONS_DIALOG_ACTIONS.DELETE]: {
+    action: COLLECTIONS_DIALOG_ACTIONS.DELETE,
+    title: 'Delete Collection',
+    description: 'Are you sure you want to delete this collection?',
+    submitButton: 'Delete',
+    showCancelButton: true,
+    showTextInput: false,
+    useDupLabel: false,
+  },
+  [COLLECTIONS_DIALOG_ACTIONS.DELETE_ERROR]: {
+    action: COLLECTIONS_DIALOG_ACTIONS.DELETE_ERROR,
+    title: 'Delete Error',
+    description: 'Cannot delete last collection.',
+    submitButton: 'OK',
+    showCancelButton: false,
+    showTextInput: false,
+    useDupLabel: false,
+  },
+};
+
+const UNIQUE_ERROR = 'Collection name must be unique.';
+const LABEL_LENGTH_ERROR = 'Collection name must be at least 1 character';
+
+const CollectionsDialog = ({
+  collectionId,
+  collectionsDialogText,
+  setCollectionsDialogText,
+  collectionLabel,
+  deleteCollectionAction,
+  renameCollectionAction,
+  duplicateCollectionAction,
+  collectionsLabels,
+  createCollectionAction,
+}) => {
+  const [inputText, setInputText] = useState('');
+  const [errorText, setErrorText] = useState('');
+
+  const {
+    title, description, submitButton, showTextInput, showCancelButton,
+  } = collectionsDialogText;
+
+  const getDefaultValue = () => {
+    if (collectionsDialogText.useDupLabel) {
+      if (collectionsDialogText.action === COLLECTIONS_DIALOG_ACTIONS.RENAME) {
+        return collectionLabel;
+      }
+      // collectionsDialogText.action === 'duplicate'
+      return `${collectionLabel} copy`;
+    }
+    return '';
+  };
+
+  useEffect(() => {
+    setInputText(getDefaultValue());
+  }, []);
+
+  const isUniqueName = ({ text, isRename, label }) => {
+    // if action is rename, new label can be same as old label
+    if (isRename && (text.toLowerCase() === label.toLowerCase())) {
+      return true;
+    }
+    return !collectionsLabels.includes(text.toLowerCase());
+  };
+
+  const hasMinLength = (text) => text.length > 0;
+
+  const hasInputErrors = ({ text, isRename, label }) => {
+    if (!hasMinLength(text)) {
+      setErrorText(LABEL_LENGTH_ERROR);
+      return true;
+    }
+    if (!isUniqueName({ text, isRename, label })) {
+      setErrorText(UNIQUE_ERROR);
```

Comment:
I wonder if on change the error should disappear
(without checking for uniqueness on every keypress)